models/src/main/scala/coop/rchain/models/rholangn/parmanager/RhoRecWriter.scala

```diff
@@ -0,0 +1,40 @@
+package coop.rchain.models.rholangn.parmanager
+
+import cats.Applicative
+import cats.syntax.all._
+import coop.rchain.models.rholangn.RhoTypeN
+import coop.rchain.models.rholangn.parmanager.primitive.PrimitiveWriter
+
+object RhoRecWriter {
+  def apply[F[_]: Applicative](
+      writer: PrimitiveWriter[F],
+      rec: RhoTypeN => F[Unit]
+  ): RhoRecWriter[F] = new RhoRecWriter(writer, rec)
+}
+
+/** Wrapper for protobuf serialization with recursive function. */
+class RhoRecWriter[F[_]: Applicative] private (
+    writer: PrimitiveWriter[F],
+    rec: RhoTypeN => F[Unit]
+) {
```

Comment:
I believe this is not specific to protobuf

Reply:
I'll update the comments.